src/core/ledger/currency.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+
+import {
+  ethAddressParser,
+  type EthAddress,
+} from "../../plugins/ethereum/ethAddress";
+import * as C from "../../util/combo";
+
+/**
+ * EvmChainId is represented in number form for all EVM-based chains,
+ * including mainnet (1), and xDai (100)
+ * The reason for this is that ethereum's client configuration utilizes
+ * a number to represent chainId, and this way we can just transpose that
+ * chainId here as a component of the currency Id, since the web3 client will
+ * return a stringified integer when the chainId is requested.
+ */
+export type EvmChainId = string;
+
+export function parseEvmChainId(id: string): EvmChainId {
+  const result = parseInt(id, 10).toString();
+  if (Number.isNaN(result) || result !== id) {
+    throw new Error(`Invalid EVM chainId value: ${id}`);
+  }
+  return id;
+}
+
+export const evmChainParser: C.Parser<EvmChainId> = C.fmap(
+  C.string,
+  parseEvmChainId
+);
+
+declare type EvmId = {
+  chainId: EvmChainId,
+  /**
+   * CurrencyAddress is a subset of all available EthAddresses.
+   *
+   * A currency address is the address of the token contract for an ERC20 token,
+   * or the 20 byte-length equivalent of 0x0, which is the conventional address
```

Comment:
Can we export the 20-byte equivalent of 0x as a constant?

Reply:
Marked resolved but not done? Is there a reason this is a bad idea? (n.b. when you resolve a comment please leave a comment explaining what the resolution is)

EDIT: i see this was added in another file. request to explain what a comment's resolution is stands. (e.g. "done, added in other file")

Reply:
Thanks, will always add resolution notes going forward.